components/eamxx/src/physics/spa/spa_functions.hpp

```diff
@@ -124,6 +127,33 @@ struct SPAFunctions
     SPAData         data;         // All spa fields
   }; // SPAInput
 
+  struct IOPReader {
+    IOPReader (iop_ptr_type& iop_,
+               const std::string file_name_,
+               const std::vector<Field>& io_fields_,
+               const std::shared_ptr<const AbstractGrid>& io_grid_)
+      : iop(iop_), file_name(file_name_)
+    {
+      field_mgr = std::make_shared<FieldManager>(io_grid_);
+      for (auto& f : io_fields_) {
+        field_mgr->add_field(f);
+        field_names.push_back(f.name());
+      }
+
+      // Set IO info for this grid and file in IOP object
+      iop->setup_io_info(file_name, io_grid_);
+    }
+
+    void read_variables(const int time_index) {
+      iop->read_fields_from_file_for_iop(file_name, field_names, util::TimeStamp(), field_mgr, time_index);
```

Comment:
I guess this is the case where the timestamp is invalid?

Reply:
You could make SPA (the atm proc) pass the current year to `update_spa_data_from_file`, and then we can create the timestamp, since `time` in the spa data is the days since the start of the year:
```
TimeStamp t({yy,1,1},{0,0,0}); // yy is the year passed by SPA
auto time = scorpio::read_time_at_index_c2f(filename,time_index);
t += time*86400;
```
Up to you. This way you'd remove the check on the timestamp in IOP. Not sure if you think it's worth it though.

Reply:
Everywhere we call this function the timestamp is available, so I've changed it so that it is just passed in.